### PR TITLE
WooCommerce: Add a nav bar to the settings screen

### PR DESCRIPTION
**Pull request by jameskoster**
This PR moves the settings navigation items from the sidebar, to a nav bar on the settings screen.

It also makes sure that the correct active item is highlighted on the sidebar when you are on subsections. For example, 'Products' is still highlighted if you are on 'Product Add' or 'Product Edit', or the list. 'Settings' is highlighted when you are on 'Payments', etc.

Desktop:

screen shot 2017-06-30 at 6 22 50 am

Mobile:

screen shot 2017-06-30 at 6 22 36 am

To Test:
* Go to `http://calypso.localhost:3000/store/settings/payments/:site`
* Make sure the nav bar shows up, and you can switch between the sections 
* Make sure that 'settings' on the sidebar is highlighted
* Go to mobile widths, and make sure the nav bar shows the correct page title, and you can select navigation options from the dropdown.
* Finally, go to product add (or another sub page) and make sure the correct section is highlighted.

### CODE REVIEW

File: client/extensions/woocommerce/app/settings/navigation.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+	siteSlug,
+	translate,
+} ) => {
+	const pathSuffix = siteSlug ? '/' + siteSlug : '';
+	const filters = [];
+
+	filters.push( {
+		id: 'payments',
+		route: '/store/settings/payments' + pathSuffix,
+		title: translate( 'Payments' ),
```
Comment (a8ci18n) [inline client/extensions/woocommerce/app/settings/navigation.js]: Hi! I've found a possible matching string that has already been translated **43** times:
`translate( 'Payment' )` *ES Score: 10*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/extensions/woocommerce/app/settings/navigation.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+export const SettingsNavigation = ( {
+	site,
+	activeSection,
+	translate,
+} ) => {
+	const items = [
+		{
+			id: 'payments',
+			path: '/store/settings/payments/:site',
+			title: translate( 'Payments' ),
```
Comment (a8ci18n) [inline client/extensions/woocommerce/app/settings/navigation.js]: Hi! I've found a possible matching string that has already been translated **43** times:
`translate( 'Payment' )` *ES Score: 10*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by justinshreve** [comment]
@kellychoffman @jameskoster I ended up handling the sidebar active states here instead of a separate PR since this hasn't been reviewed yet.

**Comment by kellychoffman** [comment]
Thanks @justinshreve. Is there any way to get the current page to show up in the mobile header? Right now its blank:

**Comment by kellychoffman** [comment]
:shipit:

**Comment by belcherj** [comment]
image

**Pull request merged**